File: projet_groupe4_back/src/main/java/barJeux/model/Avis.java
package barJeux.model;

public class Avis {
	
	private Integer id;
	private int note;
	private String titre;
	private String commentaire;
	private Reservation reservation;
	
	// Constructors 
	public Avis(Integer id, int note, String titre, String commentaire, Reservation reservation) {
		this.id = id;
		this.note = note;
		this.titre = titre;
		this.commentaire = commentaire;
		this.reservation = reservation;
	}
	public Avis(int note, String titre, String commentaire, Reservation reservation) {
		this.note = note;
		this.titre = titre;
		this.commentaire = commentaire;
		this.reservation = reservation;
	}
	
	
	// Getters & Setters
	public Integer getId() {
		return id;
	}
	public void setId(Integer id) {
		this.id = id;
	}
	public int getNote() {
		return note;
	}
	public void setNote(int note) {
		if(note < 0) note = 0;
        if(note > 5) note = 5;
        this.note = note;
	}
	public String getTitre() {
		return titre;
	}
	public void setTitre(String titre) {
		this.titre = titre;
	}
	public String getCommentaire() {
		return commentaire;
	}
	public void setCommentaire(String commentaire) {
		this.commentaire = commentaire;
	}
	public Reservation getReservation() {
		return reservation;
	}
	public void setReservation(Reservation reservation) {
		this.reservation = reservation;
	}
	// Méthodes utilitaires 


	
	// toString
	@Override
	public String toString() {
		return "Avis [id=" + id + ", note=" + note +  ", titre=" + titre + ", commentaire=" + commentaire + ", client=" + reservation.getClient().getNom() + " " + reservation.getClient().getPrenom() 
				+ ", jeu=" + reservation.getJeu().getNom() + "]";
	}
	
	
	

}

File: projet_groupe4_angular/src/app/dto/avis.ts
import { Reservation } from "./reservation";

export class Avis {
    constructor(
        public _id: number,
        public _note:number,
        public _titre:string,
        public _commentaire?:string,
        public _reservation? : Reservation
    ){}

    public get id(): number {
        return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get note(): number {
        return this._note;
    }

    public set note(value: number) {
        this._note = value;
    }

    public get titre(): string {
        return this._titre;
    }

    public set titre(value: string) {
        this._titre = value;
    }

    public get commentaire(): string | undefined{
        return this._commentaire;
    }

    public set commentaire(value: string | undefined) {
        this._commentaire = value;
    }

    public get reservation(): Reservation | undefined{
        return this._reservation;
    }

    public set reservation(value: Reservation | undefined) {
        this._reservation = value;
    }

    public toJson():any{
        return {
            note: this._note,
            titre: this._titre,
            commentaire: this._commentaire,
            reservation: this._reservation ? this._reservation.toJson?.() ?? this._reservation : undefined
        };
    }
}
